gateway/src/test/java/io/camunda/zeebe/gateway/api/deployment/DeployResourceTest.java

```diff
@@ -31,12 +31,11 @@ public void shouldMapRequestAndResponse() {
 
     final String bpmnName = "testProcess.bpmn";
     final String dmnName = "testDecision.dmn";
-    final String tenantId = "test-tenant";
+    final String defaultTenantId = "<default>";
 
     final var builder = DeployResourceRequest.newBuilder();
     builder.addResourcesBuilder().setName(bpmnName).setContent(ByteString.copyFromUtf8("<xml/>"));
     builder.addResourcesBuilder().setName(dmnName).setContent(ByteString.copyFromUtf8("test"));
-    builder.setTenantId(tenantId);
```

Comment:
❌ We should add a test case showing that mapping an explicit tenant id also works

Reply:
This would require a bit more additional work in the runtime code, since the `multiTenancyEnabled` flag for the `tenantId` check is explicitly set to `false` [here](https://github.com/camunda/zeebe/pull/14089/files#diff-1e01c457d77316e02d305a388aabd83b95ff649914d819eb06eabdd4ccd71366R83-R85).

So currently, I can't pass an explicit `tenantId` without flipping that flag to `true`. Otherwise, I'll get an exception that I'm passing a `tenantId` when multi-tenancy is disabled.

To test for the explicit mapping of a `tenantId` I would need to find a way to expose the flag, which I think is more in the scope of #14041. To avoid double work, I would add a `TODO` in this test, referencing #14041, and also comment on that issue about the TODOs.

@korthout does this work for you?

Reply:
That's fine 👍